scripts/fuzz_opt.py

```diff
@@ -690,26 +614,29 @@ def write_commands_and_test(opts):
     bytes += wasm_size
     print('post wasm size:', wasm_size)
 
-    # run some of the pair handling handlers. we don't run them all so that we get more
-    # coverage of different wasm files (but we do run all get_commands() using ones,
-    # earlier, because those are autoreducible and we want to maximize the chance of
-    # that)
-    # however, also try our best to pick a handler that can run this
+    # first, find which handlers can even run here
     relevant_handlers = [handler for handler in testcase_handlers if not hasattr(handler, 'get_commands') and handler.can_run_on_feature_opts(FEATURE_OPTS)]
-    NUM_PAIR_HANDLERS = 2
-    if len(relevant_handlers) > 0:
-        chosen_handlers = set(random.choices(relevant_handlers, k=NUM_PAIR_HANDLERS))
-        for testcase_handler in chosen_handlers:
-            assert testcase_handler.can_run_on_feature_opts(FEATURE_OPTS)
-            assert not hasattr(testcase_handler, 'get_commands')
-            if random.random() < testcase_handler.frequency:
-                print('running testcase handler:', testcase_handler.__class__.__name__)
-                testcase_handler.increment_runs()
-
-                # let the testcase handler handle this testcase however it wants. in this case we give it
-                # the input and both wasms.
-                testcase_handler.handle_pair(input=random_input, before_wasm='a.wasm', after_wasm='b.wasm', opts=opts + FUZZ_OPTS + FEATURE_OPTS)
-                print('')
+    if len(relevant_handlers) == 0:
+        return 0
+    # filter by frequency
+    filtered_handlers = [handler for handler in relevant_handlers if random.random() < handler.frequency]
+    if len(filtered_handlers) == 0:
+        # pick at least one, to not waste the effort we put into making the wasm
+        filtered_handlers = [random.choice(relevant_handlers)]
+    # run only some of the pair handling handlers. if we ran them all all the
+    # time that would mean we have less variety in wasm files and passes run
+    # on them in the same amount of time.
+    NUM_PAIR_HANDLERS = 3
+    chosen_handlers = set(random.choices(filtered_handlers, k=NUM_PAIR_HANDLERS))
```

Comment:
It seems like this would be a little simpler if we just ran all the `filtered_handlers`. Is the goal to get through more iterations more quickly by limiting the number of handlers run?

Reply:
Yes, we get more variety this way.

Imagine that we make a wasm and pick some passes to run on it A,B,C. We can run all the handlers on it. Or in the same time we can make two different wasms, the first with A,B,C and the second with D,E,F, and run half the handlers on each. So we get more code variety in the wasm and more pass variety in what we run. That's pretty important since some of the handler work overlaps (like the Determinism handler just does the same commands as the others, most obviously). And a bug may be in just a specific pass D, or ordering of passes.

Also, creating random wasms is very fast, while running handlers is much slower, so we don't lose anything by doing this.